File: src/main/java/com/edm/edmsystem/model/Document.java
package com.edm.edmsystem.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.List;


@Data
@Entity
@AllArgsConstructor
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Document {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    private DocumentType documentType;

    private DocumentStatus documentStatus;


    private String documentNumber;

    private String correctedInvoiceNumber;

    private String documentDescription;


    private LocalDate receiptDate;

    private LocalDate saleDate;

    private LocalDate issueDate;

    private boolean splitPayment;
    private boolean isElectronicDocument;


    @OneToOne(cascade = CascadeType.ALL)
    private Company supplierCompany;


    @OneToOne(cascade = CascadeType.ALL)
    private Company recipientCompany;


    @OneToOne(cascade = CascadeType.ALL)
    private Property property;

    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private List<DocumentHistory> documentChangeHistory;

    @Embedded
    private Payment payment;
}

File: emdsystem-frontend/src/app/document/document.ts
import {Payment} from "./domain/payment/payment";
import {Company} from "./domain/company/company";
import {DocumentsType} from "./domain/document-type/documents-type.enum";

export class Document {
  constructor(
    public documentId = 0,
    public supplierCompany: Company | null = null,
    public recipientCompany: Company | null = null,
    public documentType: DocumentsType | null = null,
    public issueDate: Date | null = null,
    public saleDate: Date | null = null,
    public receiptDate: Date | null = null,
    public documentNumber = '',
    public payment: Payment | null = null
  ) {
  }
}
